router/peers.go

```diff
@@ -32,6 +33,24 @@ func NewPeers(ourself *Peer, onGC func(*Peer)) *Peers {
 		onGC:    onGC}
 }
 
+func (peers *Peers) AddNewPeerFunc(f func(*Peer)) {
```

Comment:
Do we really need all this machinery? Can't we create `Peers` with a (single) `onNew` function?